### ISSUE TITLE
`insert` should behave as `upsert`, or add `upset`?

### ISSUE THREAD

**Issue by amitaibu** [pzp1997/assoc-list#5]
I guess this one is a matter of use case, but currently if an element exists, it's being removed and the new element (with the same key) is now in the head of the list.

https://github.com/pzp1997/assoc-list/blob/63ffd69a46b7805dba9c3a9ab82c5d07a8bd57a6/src/AssocList.elm#L188-L189

So, maybe we need to add a new function that will maintain the position of the element, only change the value?

**Comment by pzp1997** [comment]
Would the `update` function accomplish what you need by chance?

**Comment by amitaibu** [comment]
Hi @pzp1997  👋 The advantage of `upsert` is that it's either `insert` OR `update`, with the caller not having to bother with checking which should happen (in a way same as core's Dict.insert is actually both an insert or update)

**Comment by pzp1997** [comment]
You could always do `Dict.update key (always (Just value)) dict` but I agree that we can make this more obvious by adding an `upsert` function. I'll put it on the docket for 1.1.0.

**Comment by amitaibu** [comment]
Actually, I think  that it should be the default behavior of `insert`, as we had here:

https://github.com/Gizra/elm-dictlist/blob/ad680d1cf32504ff57aeb03f286b39f0ae676440/src/AllDictList.elm#L887-L891

**Comment by Herteby** [comment]
Yeah having it as the default behavior for `insert` seems better.
